sycl/include/CL/sycl/detail/pi.h

```diff
@@ -1157,12 +1157,15 @@ piextQueueGetNativeHandle(pi_queue queue, pi_native_handle *nativeHandle);
 ///
 /// \param nativeHandle is the native handle to create PI queue from.
 /// \param context is the PI context of the queue.
+/// \param device is the PI device associated with the native device used when
+///   creating the native queue. This parameter is optional but some backends
+///   may fail to create the right PI queue if omitted.
 /// \param queue is the PI queue created from the native handle.
 /// \param pluginOwnsNativeHandle Indicates whether the created PI object
 ///        should take ownership of the native handle.
 __SYCL_EXPORT pi_result piextQueueCreateWithNativeHandle(
-    pi_native_handle nativeHandle, pi_context context, pi_queue *queue,
-    bool pluginOwnsNativeHandle);
+    pi_native_handle nativeHandle, pi_context context, pi_device *device,
+    pi_queue *queue, bool pluginOwnsNativeHandle);
```

Comment:
Let's make the output queue argument the last one to follow other PI API
Please also bump up the PI version at the top of this file

Reply:
Good idea! I have moved the output parameter to the end of the arguments and incremented the API version.

Reply:
It seems that reaching minor version 10 for the API requires us to increase the buffer size for the version string. This is by itself an ABI break as it changes the layout of `_pi_plugin`, so accessing the version from the runtime library can give weird results and as the function table may be shifted there is no guarantee the pointers are correct between versions.

I don't think this is necessarily a concern right now, but I think we should consider reworking this soon to prevent having to change `_pi_plugin` when the version is incremented. Especially since we will have it happen again in 2 major versions.